onnxscript/ir/_core.py

```diff
@@ -1977,9 +1977,86 @@ def insert_before(self, node: Node, new_nodes: Iterable[Node] | Node, /) -> None
         self._nodes.insert_before(node, new_nodes)
 
     def sort(self) -> None:
-        """Topologically sort the nodes in the graph."""
-        raise NotImplementedError("Not implemented yet")
-
+        # make sure the graph is a DAG
+        import heapq
+        stk = []
+        nodes = [node for node in self._nodes if isinstance(node, Node)] # screen nodes
+        depth = {node: 0 for node in nodes} # topological depth
+        adjacency_list =  {node: [] for node in nodes} # parent list
+        ind = {node: i for i, node in enumerate(nodes)} # index (priority)
```

Comment:
Could you add a short comment to explain the algorithm and what these variables are used for?